abdm_integrator/hip/views/care_contexts.py

```diff
@@ -0,0 +1,92 @@
+from django.core.cache import cache
+from django.db import transaction
+from rest_framework.response import Response
+from rest_framework.status import HTTP_200_OK, HTTP_202_ACCEPTED
+
+from abdm_integrator.const import LinkRequestStatus
+from abdm_integrator.exceptions import ABDMGatewayCallbackTimeout, ABDMGatewayError, CustomError
+from abdm_integrator.hip.const import HIPGatewayAPIPath
+from abdm_integrator.hip.exceptions import HIPError
+from abdm_integrator.hip.models import LinkCareContext, LinkRequest
+from abdm_integrator.hip.serializers.care_contexts import (
+    GatewayOnAddContextsSerializer,
+    LinkCareContextRequestSerializer,
+)
+from abdm_integrator.hip.views.base import HIPBaseView, HIPGatewayBaseView
+from abdm_integrator.utils import ABDMRequestHelper, cache_key_with_prefix, poll_for_data_in_cache
+
+
+class LinkCareContextRequest(HIPBaseView):
+
+    def post(self, request, format=None):
+        serializer = LinkCareContextRequestSerializer(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        self.check_for_already_linked(serializer.data)
```

Comment:
May be rename to 'ensure_not_already_linked` or something to make it explicit that this would raise an error if the link is already there?

Reply:
Addressed in 98069dae7cba342724f71fa52aa46055a02a908d